template/tests/README-templating.md

```diff
@@ -0,0 +1,82 @@
+# Test Scenario Templating
+
+## Introduction
+
+The tests in this directory are designed to be expanded into multiple test scenarios based on test dimensions that can be defined in a dimensions file.
+
+## Defining Test Dimensions
+
+The dimensions file currently has to be named `test-definition.yaml` and reside in the same directory as the `kuttl-test.yaml.j2` file.
+
+An example of a minimal folder structure will be given further down in this file.
+
+An example of the content for the test definition file is shown here:
+
+````yaml
+dimensions:
+  - name: spark
+    values:
+      - 3.2.1
+      - 3.2.2
+      - 3.2.3
+  - name: hadoop
+    values:
+      - 3.1.0
+      - 3.2.0
+  - name: aws
+      - abc
+      - xyz
+tests:
+  - name: spark-pi-public-s3
+    dimensions:
+      - spark
+      - hadoop
+````
+
+This file defines three dimensions for this test to be considered.
```

Comment:
I recommend removing the `aws` dimension from the example above because it's not reflected anywhere below and it will create confusion.

Reply:
I intentionally added it to show that test cases need not reference all dimensions. Together with the explanatory sentence about only referencing two dimensions below.

Reply:
okay.